crossfit/backend/torch/model.py

```diff
@@ -64,32 +66,64 @@ def get_model_output(self, all_outputs_ls, index, loader, pred_output_col) -> cu
         # importing here to avoid cyclic import error
         from crossfit.backend.torch.loader import SortedSeqLoader
 
-        out = cudf.DataFrame(index=index)
+        out_df = cudf.DataFrame(index=index)
         _index = loader.sort_column(index.values) if type(loader) is SortedSeqLoader else index
-
-        if self.model_output_type == "string":
-            all_outputs = [o for output in all_outputs_ls for o in output]
-            out[pred_output_col] = cudf.Series(data=all_outputs, index=_index)
-            del all_outputs_ls
-            del loader
-        else:
-            outputs = cp.asarray(
-                concat_and_pad_tensors(
-                    all_outputs_ls,
-                    pad_token_id=loader.pad_token_id,
-                    padding_side=loader.padding_side,
+        if isinstance(all_outputs_ls[0], dict):
```

Comment:
What's the signature of `all_outputs_ls`? That'll help me understand the behavior better. Per my understanding

1. When `all_outputs_ls` is a `List[Dict[K, V]]` then 
    - all Dicts in the List have the same `K`s.
    - we create one column per `K` by calling `_add_column_to_df(out_df, [K in all dicts], ...]`
2. Otherwise it's a `List[Primitive Data Type]` in which case we create a column with `pred_output_col`

Reply:
Your understanding is correct, should i change something to make it more clearer.

Reply:
The type hint is helpful now